File: Konfigurator-aspnetcore-Backend/Konfigurator-aspnetcore-Backend/Models/DTOs/AbstractGameMapDto.cs
namespace Konfigurator_aspnetcore_Backend.Models.DTOs
{
    public abstract class AbstractGameMapDto
    {
        public AbstractGameMapDto(
            string name,
            string username,
            MapSizeDto size,
            string tile,
            string barrier,
            string character,
            string target,
            List<VectorDto> obstacles,
            VectorDto startPos,
            VectorDto endPos)
        {
            this.name = name;
            this.username = username;
            this.size = size;
            this.tile = tile;
            this.barrier = barrier;
            this.character = character;
            this.target = target;
            this.obstacles = obstacles;
            this.startPos = startPos;
            this.endPos = endPos;
        }

        public string name { get; }
        public string username { get; }
        public MapSizeDto size { get; }
        public string tile { get; }
        public string barrier { get; }
        public string character { get; }
        public string target { get; }
        public List<VectorDto> obstacles { get; }
        public VectorDto startPos { get; }
        public VectorDto endPos { get; }
    }
}

File: Konfigurator-nestjs-Backend/konfigurator-nestjs-backend/src/dto/AbstractGameMap.dto.ts
import { ApiProperty} from '@nestjs/swagger';
import { MapSizeDto } from "./MapSize.dto";
import { VectorDto } from "./Vector.dto";

export abstract class AbstractGameMapDto {
    @ApiProperty()
    name: string;

    @ApiProperty()
    username: string;

    @ApiProperty()
    size: MapSizeDto;

    @ApiProperty()
    tile: string;

    @ApiProperty()
    barrier: string;

    @ApiProperty()
    character: string;
    
    @ApiProperty()
    target: string;

    @ApiProperty()
    obstacles: VectorDto[];

    @ApiProperty()
    startPos: VectorDto;

    @ApiProperty()
    endPos: VectorDto;
}